File: app/src/main/java/com/github/panda_io/micro_panda/builder/llvm/ir/constant/zeroinitializer.go
package constant

import (
	"fmt"

	"github.com/panda-io/micro-panda/target/llvm/ir/ir"
)

type ZeroInitializer struct {
	Typ ir.Type
}

func NewZeroInitializer(typ ir.Type) *ZeroInitializer {
	return &ZeroInitializer{Typ: typ}
}

func (c *ZeroInitializer) String() string {
	return fmt.Sprintf("%s %s", c.Type().String(), c.Ident())
}

func (c *ZeroInitializer) Type() ir.Type {
	return c.Typ
}

func (c *ZeroInitializer) Ident() string {
	return "zeroinitializer"
}

File: app/src/main/java/com/github/panda_io/micro_panda/builder/llvm/ir/constant/ZeroInitializer.java
package com.github.panda_io.micro_panda.builder.llvm.ir.constant;

import com.github.panda_io.micro_panda.builder.llvm.ir.type.Type;

public class ZeroInitializer extends Constant {
    Type type;

    public ZeroInitializer(Type type) {
        this.type = type;
    }

    public String string() {
        return String.format("%s zeroinitializer", this.type.string());
    }

    public Type getType() {
        return this.type;
    }

    public String identifier() {
        return "zeroinitializer";
    }
}
